structure.go

```diff
@@ -269,6 +274,21 @@ type Playlist interface {
 	Encode() *bytes.Buffer
 	Decode(bytes.Buffer, bool) error
 	DecodeFrom(reader io.Reader, strict bool) error
+	WithCustomDecoders([]CustomDecoder) Playlist
+	String() string
+}
+
+// Interface for decoding custom and unsupported tags
+type CustomDecoder interface {
+	TagName() string
+	Decode(line string) (CustomTag, error)
+	Segment() bool
```

Comment:
What does this do?

Reply:
Okay I see, it signals whether the tag applies to the playlist or per segment.

I think the method could have a better name which is less ambiguous.

Maybe something like PerSegment, SegmentTag, IsPlaylistTag

Reply:
That is correct. I will add comments to each method in the interface as it will be useful for anyone needing to implement it.

I agree it is a bit ambiguous, I will update to `SegmentTag`

Reply:
@tmm1 I've updated my PR per this discussion, let me know if that makes things more clear, thanks! :)

Reply:
Much better, thanks. Appreciate all the comments in the public interface methods.